zulipterminal/ui_tools/buttons.py

```diff
@@ -61,7 +63,11 @@ def update_count(self, count: int, text_color: Optional[str]=None) -> None:
             count_text = ''
         else:
             count_text = str(count)
-        self.update_widget(('unread_count', count_text), new_color)
+
+        if self.starred is True:
+            self.update_widget(('starred_count', count_text), new_color)
+        else:
+            self.update_widget(('unread_count', count_text), new_color)
```

Comment:
I think it's better not to change `TopButton` to fit the needs of `StarredButton`. The parent has to be agnostic of its children. Instead you can create an _optional theme arg_ which defaults to 'unread_count' in `__init__` and `update_count` or create `modified_update_count` in child which calls `update_count`.

Reply:
interesting! I'll look into doing that. Mind if a reach out if I have questions?

Reply:
Sure, no problem :+1:

Reply:
Just updated the PR! Everything is now handled in the child StarredButton element

Reply:
It's nice that's its now handled by `StarredButton` but its just repetition of code. There are 2 ways you could do this (by creating a variable that stores text you wan't to display which can default to `unread count` when not specified):
* **Simpler Way**: Pass `'starred count'` to `__init__`, `__init__` to `update_count` , to `update_widget`
* or from the Starred button's `update_count` call `super().update_count` and pass `'starred count'` to it.

Reply:
I'm not sure I quite follow what you mean...I added the ability to pass the count_type to the Top Button (with a default of unread_count) which the update_count uses to pass the correct words to update_widget....but that feels similar to what I had in the first place?

Reply:
It looks fine now, initially you defined `starred` in the TopButton itself and said if True do this else do that. Now it's independent of the text. A future button may use this to write `'something else'` instead of `'starred count'`.
(I meant _optional text arg_ actually, I mistyped :))

Reply:
So naming it `count_theme` makes more sense than `count_type` ?

Reply:
Oh true! That makes a lot more sense, let me change the variable name real quick